sdk/eventhub/Azure.Messaging.EventHubs/tests/Shared/TestEnvironment.cs

```diff
@@ -142,13 +158,52 @@ private static string ReadAndVerifyEnvironmentVariable(string variableName)
             return environmentValue;
         }
 
+        /// <summary>
+        ///   Reads an environment variable.
+        /// </summary>
+        ///
+        /// <param name="variableName">The name of the environment variable to read.</param>
+        ///
+        /// <returns>The value of the environment variable, if present and populated; null otherwise</returns>
+        ///
+        private static string ReadEnvironmentVariable(string variableName)
+        {
+            return Environment.GetEnvironmentVariable(variableName);
+        }
+
+        /// <summary>
+        ///   It tries to read the EVENT_HUBS_NAMESPACE_CONNECTION_STRING environment variable.
+        ///   If not found, it creates a new namespace on Azure.
+        /// </summary>
+        ///
+        /// <returns>The active Event Hubs namespace for this test run.</returns>
+        ///
+        private static EventHubScope.NamespaceProperties CreateNamespaceIfMissing()
+        {
+            if (!string.IsNullOrEmpty(s_eventHubsNamespaceConnectionString.Value))
+            {
+                return GetNamespaceFromConnectionString(s_eventHubsNamespaceConnectionString.Value);
+            }
+
+            return CreateNamespace();
+        }
+
+        /// <summary>
+        ///   It instantiates a new namespace starting from the connection string passed in.
+        /// </summary>
+        ///
+        /// <returns>The active Event Hubs namespace for this test run.</returns>
+        ///
+        private static EventHubScope.NamespaceProperties GetNamespaceFromConnectionString(string connectionString) =>
```

Comment:
We may need to do a quick and dirty parse here, to avoid the reference issue and having to do weird source sharing for just that bit of logic.   Very open to suggestions here.

Reply:
It might work for now. Eventually that might be turned into a nuget reference? 

As soon as the friendly attribute was published on nuget.

Reply:
I did something a bit unusual in order to have the hard reference in place:

```
  <ItemGroup>
    <PackageReference Include="Azure.Messaging.EventHubs">
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    <ProjectReference Include="..\..\Azure.Messaging.EventHubs\src\Azure.Messaging.EventHubs.csproj" />
  </ItemGroup>
```

This is because the package reference to ```Azure.Messaging.EventHubs``` in place in ```Azure.Messaging.EventHubs.CheckpointStore.Blobs``` was overriding the project reference in ```Azure.Messaging.EventHubs.CheckpointStore.Blobs.Tests```.

Including a package reference before the project reference did seem to be ok. I am assuming that eventually this will be replaced by the preview6 of ```Azure.Messaging.EventHubs```?

Reply:
Maybe?  I'd want to think on that.  It seems like it opens a bigger door than we would want to open for one small bit of functionality that is expected to be very stable (because the service team guarantees no breaking changes at that level)